godot-core/src/builtin/mod.rs

```diff
@@ -121,6 +127,74 @@ pub(crate) fn to_usize(i: i64) -> usize {
     i.try_into().unwrap()
 }
 
+/// Trait easing converting Rust ranges to values expected by Godot.
+///
+/// Note: Unbounded upper bounds must be represented by `i32::MAX` instead of `i64::MAX`,
+/// since Godot treats some indexes as 32-bit despite being declared `i64` in GDExtension API.
```

Comment:
It isn't public therefore docstring is fine (easier to spot in IDE).

Godot indirectly informs that one should use `i32` in docs, by providing default values. For example: https://docs.godotengine.org/en/stable/classes/class_nodepath.html#class-nodepath-method-slice